File: rust/tcl-registry/src/commands/irules/cache__expire.rs
//! `CACHE::expire` iRules command.
use crate::prelude::*;
pub fn spec() -> CommandSpec {
    CommandSpec {
        name: "CACHE::expire",
        dialects: Some(DialectSet::IRULES),
        arity: Arity::at_least(0),
hover: Some(HoverSnippet {
            summary: "Forces the document to be revalidated from the server.",
            synopsis: &["CACHE::expire"],
            snippet: "Forces the document to be revalidated from the server.\n\nCACHE::expire\n\n     * Forces the document to be revalidated from the server.",
            source: "https://clouddocs.f5.com/api/irules/CACHE__expire.html",
            examples: "when CACHE_REQUEST {\n  if { $expire equals 1 } {\n    CACHE::expire\n    log \"cache expire\"\n  }\n}",
            return_value: "",
        }),
        event_requires: Some(EventRequires {
            client_side: false,
            server_side: false,
            transport: None,
            profiles: &["CACHE"],
            also_in: &[],
            init_only: false,
            flow: false,
            capability: None,
        }),
        ..CommandSpec::DEFAULT
    }
}

File: dialects/f5/irules/cache__expire.py
# tcl-lsp — a language server and toolchain for Tcl
# Copyright (C) 2026 James Deucker (bitwisecook) <https://github.com/bitwisecook>
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU Affero General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU Affero General Public License for more details.
#
# You should have received a copy of the GNU Affero General Public License
# along with this program.  If not, see <https://www.gnu.org/licenses/>.
#
# SPDX-License-Identifier: AGPL-3.0-or-later

# Enriched from F5 iRules reference documentation.
"""CACHE::expire -- Forces the document to be revalidated from the server."""

from __future__ import annotations

from compiler.registry._base import CommandDef
from compiler.registry.models import CommandSpec, FormKind, FormSpec, HoverSnippet, ValidationSpec
from compiler.registry.namespace_models import EventRequires
from compiler.registry.signatures import Arity
from compiler.side_effects import ConnectionSide, SideEffect, SideEffectTarget

from ._base import _IRULES_ONLY, register

_SOURCE = "https://clouddocs.f5.com/api/irules/CACHE__expire.html"


@register
class CacheExpireCommand(CommandDef):
    name = "CACHE::expire"

    @classmethod
    def spec(cls) -> CommandSpec:
        return CommandSpec(
            name="CACHE::expire",
            dialects=_IRULES_ONLY,
            hover=HoverSnippet(
                summary="Forces the document to be revalidated from the server.",
                synopsis=("CACHE::expire",),
                snippet=(
                    "Forces the document to be revalidated from the server.\n"
                    "\n"
                    "CACHE::expire\n"
                    "\n"
                    "     * Forces the document to be revalidated from the server."
                ),
                source=_SOURCE,
                examples=(
                    "when CACHE_REQUEST {\n"
                    "  if { $expire equals 1 } {\n"
                    "    CACHE::expire\n"
                    '    log "cache expire"\n'
                    "  }\n"
                    "}"
                ),
            ),
            forms=(
                FormSpec(
                    kind=FormKind.DEFAULT,
                    synopsis="CACHE::expire",
                ),
            ),
            validation=ValidationSpec(
                arity=Arity(),
            ),
            event_requires=EventRequires(profiles=frozenset({"CACHE"})),
            side_effect_hints=(
                SideEffect(
                    target=SideEffectTarget.STREAM_PROFILE,
                    writes=True,
                    connection_side=ConnectionSide.BOTH,
                ),
            ),
        )
